tests/protocol/test_connection.py

```diff
@@ -26,3 +26,93 @@ def test_kill(self):
     def test_ping(self):
         self.send_request("ping")
         self.assertEqualResponse("OK")
+
+    def test_tagtypes(self):
+        self.send_request("tagtypes")
+        self.assertInResponse("tagtype: Artist")
+        self.assertInResponse("tagtype: ArtistSort")
+        self.assertInResponse("tagtype: Album")
+        self.assertInResponse("tagtype: AlbumArtist")
+        self.assertInResponse("tagtype: AlbumArtistSort")
+        self.assertInResponse("tagtype: Title")
+        self.assertInResponse("tagtype: Track")
+        self.assertInResponse("tagtype: Name")
+        self.assertInResponse("tagtype: Genre")
+        self.assertInResponse("tagtype: Date")
+        self.assertInResponse("tagtype: Composer")
+        self.assertInResponse("tagtype: Performer")
+        self.assertInResponse("tagtype: Disc")
+        self.assertInResponse("tagtype: MUSICBRAINZ_ARTISTID")
+        self.assertInResponse("tagtype: MUSICBRAINZ_ALBUMID")
+        self.assertInResponse("tagtype: MUSICBRAINZ_ALBUMARTISTID")
+        self.assertInResponse("tagtype: MUSICBRAINZ_TRACKID")
+        self.assertInResponse("OK")
+
+    def test_tagtypes_clear(self):
+        self.send_request("tagtypes clear")
+        self.assertEqualResponse("OK")
+
+    def test_tagtypes_all(self):
```

Comment:
This test is only as good as the tagtypes you remember to check, and I think you are missing the new one you added: "Comment". Better to check there's a response line for everything in `TAGTYPE_LIST` ? Or count them, or something.

Reply:
I added a check to ensure the length of the response is the same as the tagtype list (plus 1 for the "OK" line).